core/ir/opnd_api.h

```diff
@@ -1307,6 +1307,268 @@ enum {
+    DR_REG_V18,
+    DR_REG_V19,
+    DR_REG_V20,
+    DR_REG_V21,
+    DR_REG_V22,
+    DR_REG_V23,
+    DR_REG_V24,
+    DR_REG_V25,
+    DR_REG_V26,
+    DR_REG_V27,
+    DR_REG_V28,
+    DR_REG_V29,
+    DR_REG_V30,
+    DR_REG_V31,
+    DR_REG_V32,
+    DR_REG_V33,
+    DR_REG_V34,
+    DR_REG_V35,
+    DR_REG_V36,
+    DR_REG_V37,
+    DR_REG_V38,
+    DR_REG_V39,
+    DR_REG_V40,
+    DR_REG_V41,
+    DR_REG_V42,
+    DR_REG_V43,
+    DR_REG_V44,
+    DR_REG_V45,
+    DR_REG_V46,
+    DR_REG_V47,
+    DR_REG_V48,
+    DR_REG_V49,
+    DR_REG_V50,
+    DR_REG_V51,
+    DR_REG_V52,
+    DR_REG_V53,
+    DR_REG_V54,
+    DR_REG_V55,
+    DR_REG_V56,
+    DR_REG_V57,
+    DR_REG_V58,
+    DR_REG_V59,
+    DR_REG_V60,
+    DR_REG_V61,
+    DR_REG_V62,
+    DR_REG_V63,
+    DR_REG_V64,
+    DR_REG_V65,
+    DR_REG_V66,
+    DR_REG_V67,
+    DR_REG_V68,
+    DR_REG_V69,
+    DR_REG_V70,
+    DR_REG_V71,
+    DR_REG_V72,
+    DR_REG_V73,
+    DR_REG_V74,
+    DR_REG_V75,
+    DR_REG_V76,
+    DR_REG_V77,
+    DR_REG_V78,
+    DR_REG_V79,
+    DR_REG_V80,
+    DR_REG_V81,
+    DR_REG_V82,
+    DR_REG_V83,
+    DR_REG_V84,
+    DR_REG_V85,
+    DR_REG_V86,
+    DR_REG_V87,
+    DR_REG_V88,
+    DR_REG_V89,
+    DR_REG_V90,
+    DR_REG_V91,
+    DR_REG_V92,
+    DR_REG_V93,
+    DR_REG_V94,
+    DR_REG_V95,
+    DR_REG_V96,
+    DR_REG_V97,
+    DR_REG_V98,
+    DR_REG_V99,
+    DR_REG_V100,
+    DR_REG_V101,
+    DR_REG_V102,
+    DR_REG_V103,
+    DR_REG_V104,
+    DR_REG_V105,
+    DR_REG_V106,
+    DR_REG_V107,
+    DR_REG_V108,
+    DR_REG_V109,
+    DR_REG_V110,
+    DR_REG_V111,
+    DR_REG_V112,
+    DR_REG_V113,
+    DR_REG_V114,
+    DR_REG_V115,
+    DR_REG_V116,
+    DR_REG_V117,
+    DR_REG_V118,
+    DR_REG_V119,
+    DR_REG_V120,
+    DR_REG_V121,
+    DR_REG_V122,
+    DR_REG_V123,
+    DR_REG_V124,
+    DR_REG_V125,
+    DR_REG_V126,
+    DR_REG_V127,
+    DR_REG_V128,
+    DR_REG_V129,
+    DR_REG_V130,
+    DR_REG_V131,
+    DR_REG_V132,
+    DR_REG_V133,
+    DR_REG_V134,
+    DR_REG_V135,
+    DR_REG_V136,
+    DR_REG_V137,
+    DR_REG_V138,
+    DR_REG_V139,
+    DR_REG_V140,
+    DR_REG_V141,
+    DR_REG_V142,
+    DR_REG_V143,
+    DR_REG_V144,
+    DR_REG_V145,
+    DR_REG_V146,
+    DR_REG_V147,
+    DR_REG_V148,
+    DR_REG_V149,
+    DR_REG_V150,
+    DR_REG_V151,
+    DR_REG_V152,
+    DR_REG_V153,
+    DR_REG_V154,
+    DR_REG_V155,
+    DR_REG_V156,
+    DR_REG_V157,
+    DR_REG_V158,
+    DR_REG_V159,
+    DR_REG_V160,
+    DR_REG_V161,
+    DR_REG_V162,
+    DR_REG_V163,
+    DR_REG_V164,
+    DR_REG_V165,
+    DR_REG_V166,
+    DR_REG_V167,
+    DR_REG_V168,
+    DR_REG_V169,
+    DR_REG_V170,
+    DR_REG_V171,
+    DR_REG_V172,
+    DR_REG_V173,
+    DR_REG_V174,
+    DR_REG_V175,
+    DR_REG_V176,
+    DR_REG_V177,
+    DR_REG_V178,
+    DR_REG_V179,
+    DR_REG_V180,
+    DR_REG_V181,
+    DR_REG_V182,
+    DR_REG_V183,
+    DR_REG_V184,
+    DR_REG_V185 = 188, /* Skip x86 DR_REG_INVALID == 187. */
```

Comment:
Let's add an assert at the top of arch_init() that we got these values right: I guess a loop over the enum values ensuring none equals _NULL or _INVALID, probably under DOCHECK level 1 or 2.

Reply:
Sounds good.
I was wondering if we had a place where we do some "sanity checks", good to know it's `d_r_arch_init()`.

Reply:
> I guess a loop over the enum values ensuring none equals _NULL or _INVALID, probably under DOCHECK level 1 or 2.

Actually, we have a gap in the enum values (because of x86 DR_REG_INVALID == 187), so we can't really loop through them. I "unrolled" the loop with all DR_REG_V values, but I don't know if it's acceptable.
Perhaps we want to move that "monstrosity" into the tests in suite/tests/api/isa_regdeps.c?
Alternatively we could just check DR_REG_V0 and DR_REG_185 to be != than _NULL or _INVALID?

Reply:
https://github.com/DynamoRIO/dynamorio/pull/6783#discussion_r1575498611